packages/providers/src/lib/sms/bulk-sms/bulk-sms.provider.ts

```diff
@@ -22,10 +23,14 @@ export class BulkSmsProvider extends BaseProvider implements ISmsProvider {
     options: ISmsOptions,
     bridgeProviderData: WithPassthrough<Record<string, unknown>> = {}
   ): Promise<ISendMessageSuccessResponse> {
+    const from = options.from || this.config.from;
+
     const payload = this.transform(bridgeProviderData, {
       to: options.to,
       body: options.content,
-      from: options.from || null,
+      ...(this.createFormField(from) && { from: this.createFormField(from) }),
+      // this userSuppliedId helps bulk-sms to identify the message source as Novuand helps in debugging
+      userSuppliedId: 'BLKTM.NOVU.01.00.00',
```

Comment:
But isn't this id needs to be unique to each message? We usually forward the message id from our mongodb id, like on sendgrid. this way we can also track unique delivery and opens, etc...

Reply:
@scopsy 

userSuppliedId was shared by BulkSMS Integration developer, check [this thread](https://app.plain.com/workspace/w_01HE2X2JVN730ZBA6QZ59X8BZ8/thread/th_01K4CFM51KD64E07H6714TFV0Q/)

Using this id, they will be able to track the source of sms message